couchbase/datadog_checks/couchbase/couchbase.py

```diff
@@ -402,3 +410,83 @@ def extract_seconds_value(self, value):
             unit = 'us'
 
         return float(val) / TO_SECONDS[unit]
+
+    def _collect_index_stats_metrics(self):
+        url = urljoin(self._index_stats_url, INDEX_STATS_METRICS_PATH)
+        try:
+            data = self._get_stats(url)
+        except requests.exceptions.RequestException as e:
+            msg = "Error accessing the Index Statistics endpoint: %s: %s" % (url, str(e))
+            self.log.warning(str(e))
+            self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.CRITICAL, self._tags, msg)
+            try:
+                # Error handling in case Couchbase changes their versioning format
+                if self._version and int(self._version.split(".")[0]) < 7:
+                    self.log.warning(
+                        "Index Stats Metrics are only available in Couchbase version 7+. Detected version: %",
```

Comment:
Since this option is behind a flag that must be enabled by the user and we properly documented the requirement, I don't know if we need this warning log. We could consider just validating the version in `check()`  before we call `_collect_index_stats_metrics`, and if version is incompatible, avoid collecting completely versus running into a requests error and log a debug line.

Elastic and postgres have examples of checking versions before collecting a subset of metrics

Reply:
I think I initially approached it in a way that we should try to get the index stats anyways in case the version logic would fail or if for whatever reason Couchbase decides to change the way they expose their version.

I moved the logic to check for index_stats_url and version and if version is greater than or equal to 7 in the check function.